spark/src/main/java/org/apache/zeppelin/spark/dep/Repository.java

```diff
@@ -51,4 +53,22 @@ public String getName() {
   public String getUrl() {
     return url;
   }
+  
+  public Repository username(String username) {
+    this.username = username;
+    return this;
+  }
+  
+  public Repository password(String password) {
+    this.password = password;
+    return this;
+  }
+  
+  protected Authentication getAuthentication() {
+    Authentication auth = null;
+    if (this.username != null && this.password != null) {
+      auth = new Authentication(this.username, this.password);
```

Comment:
It's possible password could be null or "" in some enterprise/internal authentication scheme perhaps? shouldn't we support `if this.username != null`